File: StaticSharpWeb/BasicComponents/Reactive/Math.js
function Max(a,b) {
    if (a == undefined) return b
    if (b == undefined) return a
    return Math.max(a,b)
}

function Min(a, b) {
    if (a == undefined) return b
    if (b == undefined) return a
    return Math.min(a, b)
}

function Sum() {
    let resultValid = false
    let result = 0
    for (let i of arguments) {
        if (isNaN(i)) continue;
        resultValid = true
        result += i
    }
    return resultValid ? result : undefined
}

File: StaticSharp/CrossplatformLibrary/Math/Math.cs
using Javascriptifier;

namespace StaticSharp {

    namespace Scripts {
        public class MathAttribute : ScriptReferenceAttribute {
            public MathAttribute() : base(GetScriptFilePath()) { }
        }
    }




    namespace Js {

        public static class Constants {
            public static readonly string Undefined = "undefined";
        }


        [JavascriptClass("")]
        public static class Math {
            [JavascriptOnlyMember]
            public static double First(params double[] value) => throw new JavascriptOnlyException();

            [JavascriptOnlyMember]
            public static double Sum(params double[] value) => throw new JavascriptOnlyException();

            [JavascriptOnlyMember]
            public static double Min(params double[] value) => throw new JavascriptOnlyException();

            [JavascriptOnlyMember]
            public static double Max(params double[] value) => throw new JavascriptOnlyException();
            
            [JavascriptOnlyMember]
            public static double Clamp(double value, double min, double max ) => throw new JavascriptOnlyException();
        }

    }


}

